src/commonControls/numberControl/NumberControl.ts

```diff
@@ -267,6 +298,11 @@ export class NumberControlState implements ControlState {
      * Tracks the values the user disconfirmed.
      */
     rejectedValues: number[] = [];
+
+    /**
+     * Tracks whether the value is valid.
+     */
+    isValidValue: boolean = true;
```

Comment:
Please remove and  recalculate each time as the validation function may take dependency on non-local things.  (also, general rule is to only track state that cannot be recalculated)

Example of non-local function:  validation: ()=> this.number > someOtherControl.value